src/Strongly.Options.SourceGenerators/Extensions/AssemblySymbolExtensions.cs

```diff
@@ -0,0 +1,33 @@
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Strongly.Options.SourceGenerators.Extensions;
+
+using static WellKnownNamings;
+
+public static class AssemblySymbolExtensions
+{
+    public static string GetModuleNameFromAttributeOrShortAssemblyNameByDefault(this IAssemblySymbol assembly)
+    {
+        var userSpecifiedModuleName = assembly
+           .GetAttributes()
+           .FirstOrDefault(x => x.AttributeClass!.ToDisplayString() == StronglyOptionsModuleAttribute)?
+           .ConstructorArguments
+           .FirstOrDefault()
+           .Value?
+           .ToString();
+
+        return userSpecifiedModuleName ?? assembly.GetLastAssemblyNamePart();
+    }
+
+    private static string GetLastAssemblyNamePart(this IAssemblySymbol assembly)
+    {
+        var assemblyName = assembly.Name;
+        var lastIndexOfDot = assemblyName.LastIndexOf('.');
```

Comment:
```suggestion
        var indexOfLastDot = assemblyName.LastIndexOf('.');
```
последний индекс точки как-то по татарски звучит